samples/QuickstartClient/Program.cs

```diff
@@ -89,6 +90,16 @@ static void PromptForInput()
         [var script] when script.EndsWith(".py") => ("python", args),
         [var script] when script.EndsWith(".js") => ("node", args),
         [var script] when Directory.Exists(script) || (File.Exists(script) && script.EndsWith(".csproj")) => ("dotnet", ["run", "--project", script]),
-        _ => ("dotnet", ["run", "--project", "../QuickstartWeatherServer"])
+        _ => ("dotnet", ["run", "--project", Path.Combine(GetCurrentSourceDirectory(), "../QuickstartWeatherServer")])
     };
+}
+
+static string GetCurrentSourceDirectory([CallerFilePath] string? currentFile = null)
+{
+    if (string.IsNullOrWhiteSpace(currentFile))
```

Comment:
If this is null or whitespace, it's a compiler bug. If you really want to validate this, you could assert.

Reply:
Hence an assert.